### ISSUE TITLE
Error when installing

### ISSUE THREAD

**Issue by fearwillkeepus** [RobotsOnDrugs/Moo.FuckScreenConnect-rs#1]
I am getting the following error when running the install ps1 in Powershell.
I have done everything noted for the prereqs.

[2024-11-04 12:00:43 UTC-07] INFO: Starting.
[2024-11-04 12:00:43 UTC-07] INFO: Installing FSC.
[2024-11-04 12:00:47 UTC-07] INFO: Installation location: C:\AppData\Logs\Users\MYPC\AppData\FSC.
[2024-11-04 12:00:47 UTC-07] INFO: Installing from scratch.
[2024-11-04 12:00:47 UTC-07] INFO: Creating a new installation.
Error[0].Message : The term 'Error[0].Message' is not recognized as the name of a cmdlet, function, script file, or
operable program. Check the spelling of the name, or if a path was included, verify that the path is correct and try
again.
At C:\AppData\Logs\Users\MYPC\AppData\FSC\install_modules\fsc_functions.ps1:69 char:64
+ ... Error -ErrorMessage "Could not copy files: $(Error[0].Message). Canno ...
+                                                  ~~~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (Error[0].Message:String) [], ParentContainsErrorRecordException
    + FullyQualifiedErrorId : CommandNotFoundException

**Comment by RobotsOnDrugs** [comment]
Ugh. Not sure how a typo got in there after testing. I hate error handling in Powershell.
On line 69 of fsc_functions, put a dollar sign before `Error` so it looks like this and it should give you the actual error:
`Stop-ScriptWithError -ErrorMessage "Could not copy files: $($Error[0].Message). Cannot continue."`
I'll put up a fix after doing some testing for that chunk of error handling. I'm not looking through the project at the moment, but I did notice that the syntax is incorrect here.

~It could be a permission error - it needs to be run as admin. Apparently, I forgot to put in a check for that, but that's a separate issue that I'll also fix.~

~If you're running as admin and it still gives you an error, paste it here.~ Also, please put any output in a code block so that it's formatted properly.

Thanks for the report.

**Comment by RobotsOnDrugs** [comment]
Did that change give you a new error? There could be a separate bug, but without knowing the error you get, I can't really investigate.

**Comment by fearwillkeepus** [comment]
After adding the $, here is the error.
`[2024-11-13 12:58:38 UTC-07] ERROR: Could not copy files: . Cannot continue.`

**Comment by RobotsOnDrugs** [comment]
Okay, so this bit of error checking did indeed get messed up somehow with the wrong stuff remaining after testing.

At line 68 in fsc_functions.ps1, Replace:
```ps1
Remove-Item -Recurse $install_path -Force -ErrorAction SilentlyContinue | Out-Null
Stop-ScriptWithError -ErrorMessage "Could not copy files: $($Error[0].Message). Cannot continue."
```
with:
```ps1
Remove-Item -Recurse $install_path -Force -ErrorAction SilentlyContinue | Out-Null
Write-Log -Level WARN -Message "Name of file to install: $file_name"
Write-Log -Level WARN -Message "Installation path: $install_path"
$Error[0].CategoryInfo | Format-List
Stop-ScriptWithError -ErrorMessage "Could not copy files: $Error[0]. Cannot continue."
```

then run it again and paste the output here.

Hopefully it'll become clear what's causing the error in the first place and it can be fixed.

**Comment by fearwillkeepus** [comment]
Okay, I tried what you said above, but I got this error:

`Write-Log : Cannot validate argument on parameter 'Level'. The argument "WARN" does not belong to the set
"DEBUG,ERROR,INFO,NOTSET,WARNING" specified by the ValidateSet attribute. Supply an argument that is in the set and
then try the command again.
At C:\AppData\Logs\Users\MYPC\Moisty\AppData\FSC\install_modules\fsc_functions.ps1:69 char:30
    +             Write-Log -Level WARN -Message "Name of file to install:  ..."
    +                              ~~~~
    + CategoryInfo          : InvalidData: (:) [Write-Log], ParentContainsErrorRecordException
    + FullyQualifiedErrorId : ParameterArgumentValidationError,Write-Log`



So I replaced 'WARN' with 'warning" and got this error:

`[2024-11-18 15:35:18 UTC-07] INFO: Starting.
[2024-11-18 15:35:18 UTC-07] INFO: Installing FSC.
[2024-11-18 15:35:20 UTC-07] INFO: Installation location: C:\AppData\Logs\Users\MYPC\Moisty\AppData\FSC.
[2024-11-18 15:35:20 UTC-07] INFO: Installing from scratch.
[2024-11-18 15:35:20 UTC-07] INFO: Creating a new installation.
[2024-11-18 15:35:20 UTC-07] WARNING: Name of file to install: moo_fuck_screen_connect.exe
[2024-11-18 15:35:20 UTC-07] WARNING: Installation path: C:\AppData\Logs\Users\MYPC\Moisty\AppData\FSC


Category   : InvalidOperation
Activity   : Remove-Item
Reason     : PSInvalidOperationException
TargetName :
TargetType :

[2024-11-18 15:09:04 UTC-07] ERROR: Could not copy files: Cannot remove the item at 'C:\AppData\Logs\Users\MYPC\Moisty\AppData\FSC' because it is in use. Cannot overwrite the item C:\AppData\Logs\Users\MYPC\Moisty\AppData\FSC\moo_fuck_screen_connect.exe with itself. Cannot find path 'C:\Users\MYPC\AppData\Local\Temp\fsc_temp' because it does not exist. Cannot remove the item at 'C:\AppData\Logs\Users\MYPC\Moisty\AppData\FSC' because it is in use. Cannot find path 'HKLM:\SOFTWARE\Moo\FuckScreenConnect' because it does not exist. No MSFT_ScheduledTask objects found with property 'TaskName' equal to 'FSC Service Task'.  Verify the value of the property and retry. Cannot find path 'HKLM:\SOFTWARE\Moo\FuckScreenConnect' because it does not exist. No MSFT_ScheduledTask objects found with property 'TaskName' equal to 'FSC Service Task'.  Verify the value of the property and retry. Cannot find path 'HKLM:\SOFTWARE\Moo\FuckScreenConnect' because it does not exist. Cannot find path 'HKLM:\SOFTWARE\Moo\FuckScreenConnect' because it does not exist. Cannot remove the item at 'C:\AppData\Logs\Users\MYPC\Moisty\AppData\FSC' because it is in use. Cannot overwrite the item C:\AppData\Logs\Users\MYPC\Moisty\AppData\FSC\moo_fuck_screen_connect.exe with itself. Cannot find path 'C:\Users\MYPC\AppData\Local\Temp\fsc_temp' because it does not exist. Cannot remove the item at 'C:\AppData\Logs\Users\MYPC\Moisty\AppData\FSC' because it is in use. Cannot find path 'HKLM:\SOFTWARE\Moo\FuckScreenConnect' because it does not exist. No MSFT_ScheduledTask objects found with property 'TaskName' equal to 'FSC Service Task'.  Verify the value of the property and retry. Cannot find path 'HKLM:\SOFTWARE\Moo\FuckScreenConnect' because it does not exist. No MSFT_ScheduledTask objects found with property 'TaskName' equal to 'FSC Service Task'.  Verify the value of the property and retry. Cannot find path 'HKLM:\SOFTWARE\Moo\FuckScreenConnect' because it does not exist. Cannot find path 'HKLM:\SOFTWARE\Moo\FuckScreenConnect' because it does not exist. Cannot remove the item at 'C:\AppData\Logs\Users\MYPC\Moisty\AppData\FSC' because it is in use. Cannot find path 'C:\AppData\Logs\Users\MYPC\Moisty\AppData\FSC\moo_fuck_screen_connect.exe' because it does not exist. Cannot find path 'C:\Users\MYPC\AppData\Local\Temp\fsc_temp' because it does not exist. Cannot remove the item at 'C:\AppData\Logs\Users\MYPC\Moisty\AppData\FSC' because it is in use. Cannot find path 'HKLM:\SOFTWARE\Moo\FuckScreenConnect' because it does not exist. No MSFT_ScheduledTask objects found with property 'TaskName' equal to 'FSC Service Task'.  Verify the value of the property and retry. Cannot find path 'HKLM:\SOFTWARE\Moo\FuckScreenConnect' because it does not exist. No MSFT_ScheduledTask objects found with property 'TaskName' equal to 'FSC Service Task'.  Verify the value of the property and retry. Cannot find path 'HKLM:\SOFTWARE\Moo\FuckScreenConnect' because it does not exist. Cannot find path 'HKLM:\SOFTWARE\Moo\FuckScreenConnect' because it does not exist. Cannot remove the item at 'C:\AppData\Logs\Users\MYPC\Moisty\AppData\FSC' because it is in use. Cannot find path 'C:\AppData\Logs\Users\MYPC\Moisty\AppData\FSC\moo_fuck_screen_connect.exe' because it does not exist. Cannot find path 'C:\Users\MYPC\AppData\Local\Temp\fsc_temp' because it does not exist. Cannot remove the item at 'C:\AppData\Logs\Users\MYPC\Moisty\AppData\FSC' because it is in use. Cannot find path 'HKLM:\SOFTWARE\Moo\FuckScreenConnect' because it does not exist. No MSFT_ScheduledTask objects found with property 'TaskName' equal to 'FSC Service Task'.  Verify the value of the property and retry. Cannot find path 'HKLM:\SOFTWARE\Moo\FuckScreenConnect' because it does not exist. No MSFT_ScheduledTask objects found with property 'TaskName' equal to 'FSC Service Task'.  Verify the value of the property and retry. Cannot find path 'HKLM:\SOFTWARE\Moo\FuckScreenConnect' because it does not exist. Cannot find path 'HKLM:\SOFTWARE\Moo\FuckScreenConnect' because it does not exist. Cannot remove the item at 'C:\AppData\Logs\Users\MYPC\Moisty\AppData\FSC' because it is in use. Cannot find path 'C:\AppData\Logs\Users\MYPC\Moisty\AppData\FSC\moo_fuck_screen_connect.exe' because it does not exist. Cannot find path 'C:\Users\MYPC\AppData\Local\Temp\fsc_temp' because it does not exist. Cannot remove the item at 'C:\AppData\Logs\Users\MYPC\Moisty\AppData\FSC' because it is in use. Cannot find path 'HKLM:\SOFTWARE\Moo\FuckScreenConnect' because it does not exist. No MSFT_ScheduledTask objects found with property 'TaskName' equal to 'FSC Service Task'.  Verify the value of the property and retry. Cannot find path 'HKLM:\SOFTWARE\Moo\FuckScreenConnect' because it does not exist. No MSFT_ScheduledTask objects found with property 'TaskName' equal to 'FSC Service Task'.  Verify the value of the property and retry. Cannot find path 'HKLM:\SOFTWARE\Moo\FuckScreenConnect' because it does not exist. Cannot find path 'HKLM:\SOFTWARE\Moo\FuckScreenConnect' because it does not exist. Cannot remove the item at 'C:\AppData\Logs\Users\MYPC\Moisty\AppData\FSC' because it is in use. Cannot overwrite the item C:\AppData\Logs\Users\MYPC\Moisty\AppData\FSC\moo_fuck_screen_connect.exe with itself. Cannot find path 'C:\Users\MYPC\AppData\Local\Temp\fsc_temp' because it does not exist. Cannot remove the item at 'C:\AppData\Logs\Users\MYPC\Moisty\AppData\FSC' because it is in use. Cannot find path 'HKLM:\SOFTWARE\Moo\FuckScreenConnect' because it does not exist. No MSFT_ScheduledTask objects found with property 'TaskName' equal to 'FSC Service Task'.  Verify the value of the property and retry. Cannot find path 'HKLM:\SOFTWARE\Moo\FuckScreenConnect' because it does not exist. No MSFT_ScheduledTask objects found with property 'TaskName' equal to 'FSC Service Task'.  Verify the value of the property and retry. Cannot find path 'HKLM:\SOFTWARE\Moo\FuckScreenConnect' because it does not exist. Cannot find path 'HKLM:\SOFTWARE\Moo\FuckScreenConnect' because it does not exist. Cannot remove the item at 'C:\AppData\Logs\Users\MYPC\Moisty\AppData\FSC' because it is in use. Cannot overwrite the item C:\AppData\Logs\Users\MYPC\Moisty\AppData\FSC\moo_fuck_screen_connect.exe with itself. Cannot find path 'C:\Users\MYPC\AppData\Local\Temp\fsc_temp' because it does not exist. Cannot remove the item at 'C:\AppData\Logs\Users\MYPC\Moisty\AppData\FSC' because it is in use. Cannot find path 'HKLM:\SOFTWARE\Moo\FuckScreenConnect' because it does not exist. No MSFT_ScheduledTask objects found with property 'TaskName' equal to 'FSC Service Task'.  Verify the value of the property and retry. Cannot find path 'HKLM:\SOFTWARE\Moo\FuckScreenConnect' because it does not exist. No MSFT_ScheduledTask objects found with property 'TaskName' equal to 'FSC Service Task'.  Verify the value of the property and retry. Cannot find path 'HKLM:\SOFTWARE\Moo\FuckScreenConnect' because it does not exist. Cannot find path 'HKLM:\SOFTWARE\Moo\FuckScreenConnect' because it does not exist. System.Management.Automation.ParameterBindingValidationException: Cannot validate argument on parameter 'Level'. The argument "WARN" does not belong to the set "DEBUG,ERROR,INFO,NOTSET,WARNING" specified by the ValidateSet attribute. Supply an argument that is in the set and then try the command again. ---> System.Management.Automation.ValidationMetadataException: The argument "WARN" does not belong to the set "DEBUG,ERROR,INFO,NOTSET,WARNING" specified by the ValidateSet attribute. Supply an argument that is in the set and then try the command again.
   at System.Management.Automation.ValidateSetAttribute.ValidateElement(Object element)
   at System.Management.Automation.ParameterBinderBase.BindParameter(CommandParameterInternal parameter, CompiledCommandParameter parameterMetadata, ParameterBindingFlags flags)
   --- End of inner exception stack trace ---
   at System.Management.Automation.Runspaces.PipelineBase.Invoke(IEnumerable input)
   at Microsoft.PowerShell.Executor.ExecuteCommandHelper(Pipeline tempPipeline, Exception& exceptionThrown, ExecutionOptions options) Cannot validate argument on parameter 'Level'. The argument "WARN" does not belong to the set "DEBUG,ERROR,INFO,NOTSET,WARNING" specified by the ValidateSet attribute. Supply an argument that is in the set and then try the command again. Cannot remove the item at 'C:\AppData\Logs\Users\MYPC\Moisty\AppData\FSC' because it is in use. Cannot overwrite the item C:\AppData\Logs\Users\MYPC\Moisty\AppData\FSC\moo_fuck_screen_connect.exe with itself. Cannot find path 'C:\Users\MYPC\AppData\Local\Temp\fsc_temp' because it does not exist. Cannot remove the item at 'C:\AppData\Logs\Users\MYPC\Moisty\AppData\FSC' because it is in use. Cannot find path 'HKLM:\SOFTWARE\Moo\FuckScreenConnect' because it does not exist. No MSFT_ScheduledTask objects found with property 'TaskName' equal to 'FSC Service Task'.  Verify the value of the property and retry. Cannot find path 'HKLM:\SOFTWARE\Moo\FuckScreenConnect' because it does not exist. No MSFT_ScheduledTask objects found with property 'TaskName' equal to 'FSC Service Task'.  Verify the value of the property and retry. Cannot find path 'HKLM:\SOFTWARE\Moo\FuckScreenConnect' because it does not exist. Cannot find path 'HKLM:\SOFTWARE\Moo\FuckScreenConnect' because it does not exist. System.Management.Automation.CommandNotFoundException: The term 'Error[0].Message' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the spelling of the name, or if a path was included, verify that the path is correct and try again.
   at System.Management.Automation.Runspaces.PipelineBase.Invoke(IEnumerable input)
   at Microsoft.PowerShell.Executor.ExecuteCommandHelper(Pipeline tempPipeline, Exception& exceptionThrown, ExecutionOptions options) The term 'Error[0].Message' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the spelling of the name, or if a path was included, verify that the path is correct and try again. Cannot remove the item at 'C:\AppData\Logs\Users\MYPC\Moisty\AppData\FSC' because it is in use. Cannot overwrite the item C:\AppData\Logs\Users\MYPC\Moisty\AppData\FSC\moo_fuck_screen_connect.exe with itself. Cannot find path 'C:\Users\MYPC\AppData\Local\Temp\fsc_temp' because it does not exist. Cannot remove the item at 'C:\AppData\Logs\Users\MYPC\Moisty\AppData\FSC' because it is in use. Cannot find path 'HKLM:\SOFTWARE\Moo\FuckScreenConnect' because it does not exist. No MSFT_ScheduledTask objects found with property 'TaskName' equal to 'FSC Service Task'.  Verify the value of the property and retry. Cannot find path 'HKLM:\SOFTWARE\Moo\FuckScreenConnect' because it does not exist. No MSFT_ScheduledTask objects found with property 'TaskName' equal to 'FSC Service Task'.  Verify the value of the property and retry. Cannot find path 'HKLM:\SOFTWARE\Moo\FuckScreenConnect' because it does not exist. Cannot find path 'HKLM:\SOFTWARE\Moo\FuckScreenConnect' because it does not exist. System.Management.Automation.CommandNotFoundException: The term 'Error[0].Message' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the spelling of the name, or if a path was included, verify that the path is correct and try again.
   at System.Management.Automation.Runspaces.PipelineBase.Invoke(IEnumerable input)
   at Microsoft.PowerShell.Executor.ExecuteCommandHelper(Pipeline tempPipeline, Exception& exceptionThrown, ExecutionOptions options) The term 'Error[0].Message' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the spelling of the name, or if a path was included, verify that the path is correct and try again. Cannot remove the item at 'C:\AppData\Logs\Users\MYPC\Moisty\AppData\FSC' because it is in use. Cannot overwrite the item C:\AppData\Logs\Users\MYPC\Moisty\AppData\FSC\moo_fuck_screen_connect.exe with itself. Cannot find path 'C:\Users\MYPC\AppData\Local\Temp\fsc_temp' because it does not exist. Cannot remove the item at 'C:\AppData\Logs\Users\MYPC\Moisty\AppData\FSC' because it is in use. Cannot find path 'HKLM:\SOFTWARE\Moo\FuckScreenConnect' because it does not exist. No MSFT_ScheduledTask objects found with property 'TaskName' equal to 'FSC Service Task'.  Verify the value of the property and retry. Cannot find path 'HKLM:\SOFTWARE\Moo\FuckScreenConnect' because it does not exist. No MSFT_ScheduledTask objects found with property 'TaskName' equal to 'FSC Service Task'.  Verify the value of the property and retry. Cannot find path 'HKLM:\SOFTWARE\Moo\FuckScreenConnect' because it does not exist. Cannot find path 'HKLM:\SOFTWARE\Moo\FuckScreenConnect' because it does not exist. System.Management.Automation.CommandNotFoundException: The term 'Error[0].Message' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the spelling of the name, or if a path was included, verify that the path is correct and try again.
   at System.Management.Automation.Runspaces.PipelineBase.Invoke(IEnumerable input)
   at Microsoft.PowerShell.Executor.ExecuteCommandHelper(Pipeline tempPipeline, Exception& exceptionThrown, ExecutionOptions options) The term 'Error[0].Message' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the spelling of the name, or if a path was included, verify that the path is correct and try again. Cannot remove the item at 'C:\AppData\Logs\Users\MYPC\Moisty\AppData\FSC' because it is in use. Cannot overwrite the item C:\AppData\Logs\Users\MYPC\Moisty\AppData\FSC\moo_fuck_screen_connect.exe with itself. Cannot find path 'C:\Users\MYPC\AppData\Local\Temp\fsc_temp' because it does not exist. Cannot remove the item at 'C:\AppData\Logs\Users\MYPC\Moisty\AppData\FSC' because it is in use. Cannot find path 'HKLM:\SOFTWARE\Moo\FuckScreenConnect' because it does not exist. No MSFT_ScheduledTask objects found with property 'TaskName' equal to 'FSC Service Task'.  Verify the value of the property and retry. Cannot find path 'HKLM:\SOFTWARE\Moo\FuckScreenConnect' because it does not exist. No MSFT_ScheduledTask objects found with property 'TaskName' equal to 'FSC Service Task'.  Verify the value of the property and retry. Cannot find path 'HKLM:\SOFTWARE\Moo\FuckScreenConnect' because it does not exist. Cannot find path 'HKLM:\SOFTWARE\Moo\FuckScreenConnect' because it does not exist. System.Management.Automation.CommandNotFoundException: The term 'Error[0].Message' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the spelling of the name, or if a path was included, verify that the path is correct and try again.
   at System.Management.Automation.Runspaces.PipelineBase.Invoke(IEnumerable input)
   at Microsoft.PowerShell.Executor.ExecuteCommandHelper(Pipeline tempPipeline, Exception& exceptionThrown, ExecutionOptions options) The term 'Error[0].Message' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the spelling of the name, or if a path was included, verify that the path is correct and try again. Cannot remove the item at 'C:\AppData\Logs\Users\MYPC\Moisty\AppData\FSC' because it is in use. Cannot overwrite the item C:\AppData\Logs\Users\MYPC\Moisty\AppData\FSC\moo_fuck_screen_connect.exe with itself. Cannot find path 'C:\Users\MYPC\AppData\Local\Temp\fsc_temp' because it does not exist. Cannot remove the item at 'C:\AppData\Logs\Users\MYPC\Moisty\AppData\FSC' because it is in use. Cannot find path 'HKLM:\SOFTWARE\Moo\FuckScreenConnect' because it does not exist. No MSFT_ScheduledTask objects found with property 'TaskName' equal to 'FSC Service Task'.  Verify the value of the property and retry. Cannot find path 'HKLM:\SOFTWARE\Moo\FuckScreenConnect' because it does not exist. No MSFT_ScheduledTask objects found with property 'TaskName' equal to 'FSC Service Task'.  Verify the value of the property and retry. Cannot find path 'HKLM:\SOFTWARE\Moo\FuckScreenConnect' because it does not exist. Cannot find path 'HKLM:\SOFTWARE\Moo\FuckScreenConnect' because it does not exist. System.Management.Automation.CommandNotFoundException: The term 'Error[0].Message' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the spelling of the name, or if a path was included, verify that the path is correct and try again.
   at System.Management.Automation.Runspaces.PipelineBase.Invoke(IEnumerable input)
   at Microsoft.PowerShell.Executor.ExecuteCommandHelper(Pipeline tempPipeline, Exception& exceptionThrown, ExecutionOptions options) The term 'Error[0].Message' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the spelling of the name, or if a path was included, verify that the path is correct and try again. Cannot remove the item at 'C:\AppData\Logs\Users\MYPC\Moisty\AppData\FSC' because it is in use. Cannot overwrite the item C:\AppData\Logs\Users\MYPC\Moisty\AppData\FSC\moo_fuck_screen_connect.exe with itself. Cannot find path 'C:\Users\MYPC\AppData\Local\Temp\fsc_temp' because it does not exist. Cannot remove the item at 'C:\AppData\Logs\Users\MYPC\Moisty\AppData\FSC' because it is in use. Cannot find path 'HKLM:\SOFTWARE\Moo\FuckScreenConnect' because it does not exist. No MSFT_ScheduledTask objects found with property 'TaskName' equal to 'FSC Service Task'.  Verify the value of the property and retry. Cannot find path 'HKLM:\SOFTWARE\Moo\FuckScreenConnect' because it does not exist. No MSFT_ScheduledTask objects found with property 'TaskName' equal to 'FSC Service Task'.  Verify the value of the property and retry. Cannot find path 'HKLM:\SOFTWARE\Moo\FuckScreenConnect' because it does not exist. Cannot find path 'HKLM:\SOFTWARE\Moo\FuckScreenConnect' because it does not exist. Cannot find path 'C:\AppData\Logs\Users\MYPC\Moisty\AppData\FSC\.install.ps1' because it does not exist.[0]. Cannot continue.`
